analytical_engine/apps/simple_path/is_simple_path_context.h

```diff
@@ -0,0 +1,104 @@
+/** Copyright 2020 Alibaba Group Holding Limited.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef ANALYTICAL_ENGINE_APPS_SIMPLE_PATH_IS_SIMPLE_PTAH_CONTEXT_H_
+#define ANALYTICAL_ENGINE_APPS_SIMPLE_PATH_IS_SIMPLE_PTAH_CONTEXT_H_
+
+#include <limits>
+#include <utility>
+#include <vector>
+#include <set>
+
+#include "folly/dynamic.h"
+#include "folly/json.h"
+#include "grape/grape.h"
+
+
+#include "core/context/tensor_context.h"
+
+namespace gs {
+
+template <typename FRAG_T>
+class IsSimplePathContext : public TensorContext<FRAG_T, typename FRAG_T::oid_t> {
+ public:
+  using oid_t = typename FRAG_T::oid_t;
+  using vid_t = typename FRAG_T::vid_t;
+  using vertex_t = typename FRAG_T::vertex_t;
+
+  explicit IsSimplePathContext(const FRAG_T& fragment)
+      : TensorContext<FRAG_T, typename FRAG_T::oid_t>(fragment) {}
+
+  /**
+   * @brief json formate
+   *  josn = {"nodes" : [(oid_t)node1,(oid_t)node2,....]}
+   * 
```

Comment:
Maybe just use '[node1, node2, node3,....]' format would be fine and clear. folly::json can parse it to a folly::Dynamic::Array, and you can iterate the nodes like [folly](https://github.com/facebook/folly/blob/c2ea37612a79d7017db330e47cc0773423bef315/folly/test/DynamicTest.cpp#L601-L609)

Reply:
and revise the nodes process in Init base the new format.